### PR TITLE
[Fix #6597] - LineEndConcatenation is unsafe

### LINKED ISSUE #6597: Autocorrecting Style/LineEndConcatenation changes behaviour
When using `<<` to append multiple string to array, `Style/LineEndConcatenation` suggests using `\` to join the strings together.

```ruby
array = []
array << 'Foo' <<
         'Bar' <<
         'Baz'
puts array.join('-')
```

---

## Expected behavior

The way we append strings to array should not be changed. Running corrected code should output `Foo-Bar-Baz`

## Actual behavior

Autocorrect changes the code to use `\` instead of `<<` which causes strings to be joined before appending them to the array. Running corrected code outputs `FooBarBaz`

## Steps to reproduce the problem

Autocorrect the code snippet above with `rubocop -a`

## RuboCop version

```
$ rubocop -V
0.61.1 (using Parser 2.5.3.0, running on ruby 2.5.3 x86_64-linux)
```

### PR DESCRIPTION
**Pull request by jaredbeck**
* [x] Wrote [good commit messages][1].
* [x] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [x] Squashed related commits together.
* [ ] Added tests.
* [x] Added an entry to the [Changelog](../blob/master/CHANGELOG.md) if the new code introduces user-observable changes. See [changelog entry format](../blob/master/CONTRIBUTING.md#changelog-entry-format).
* [x] The PR relates to *only* one subject with a clear title
  and description in grammatically correct, complete sentences.
* [x] Run `bundle exec rake default`. It executes all tests and RuboCop for itself, and generates the documentation.

[1]: https://chris.beams.io/posts/git-commit/

### CODE REVIEW

File: config/default.yml [hunk 1]
```diff
@@ -3352,6 +3352,7 @@ Style/LineEndConcatenation:
                  Use \ instead of + or << to concatenate two string literals at
                  line end.
   Enabled: true
+  Safe: false
```
Comment (koic) [inline config/default.yml]: Maybe I think that `SafeAutoCorrect` rather than `Safe` should be specified.

```diff
-Safe: false
+SafeAutoCorrect: false
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -5,6 +5,7 @@
 ### New features
 
 * [#6580](https://github.com/rubocop-hq/rubocop/pull/6580): New cop `Rails/LinkToBlank` checks for `link_to` calls with `target: '_blank'` and no `rel: 'noopener'`. ([@Intrepidd][])
+* [#6597](https://github.com/rubocop-hq/rubocop/issues/6597): `Style/LineEndConcatenation` is now known to be un-`Safe`. ([@jaredbeck][])
```
Comment (koic) [inline CHANGELOG.md]: It's a change in behavior so write it in the `### Changes` section.

**Comment by jaredbeck** [comment]
@koic changes made, thanks!

**Comment by Drenmi** [comment]
@jaredbeck This change is good to go. Just needs a rebase and regenerating of the documentation. 🙇

File: config/default.yml [hunk 3]
```diff
@@ -3352,7 +3352,9 @@ Style/LineEndConcatenation:
                  Use \ instead of + or << to concatenate two string literals at
                  line end.
   Enabled: true
+  SafeAutoCorrect: false
   VersionAdded: '0.18'
+  VersionChanged: '0.63'
```
Comment (koic) [inline config/default.yml]: I'm sorry I'm late about merging 💦
Please update this to `'0.64'`,  regenerate the document (Run `rake generate_cops_documentation` command) and solve CHANGELOG.md with the latest master branch.

**Comment by jaredbeck** [comment]
> I'm sorry I'm late about merging 💦 Please update this to '0.64', regenerate the document (Run rake generate_cops_documentation command) and solve CHANGELOG.md with the latest master branch.

Rebased and updated as requested, thanks Ito-san.

**Pull request merged**